leasing/importer/area.py

```diff
@@ -388,14 +389,17 @@ def execute(self):  # NOQA C901
 
                 try:
                     areas.update_or_create(defaults=other_data, **match_data)
-                except MultipleObjectsReturned:  # There should only be one object per identifier...
-                    ext_id = other_data.pop("external_id")
-                    # ...so we delete them all but spare the one with the correct external_id (if it happens to exist)
-                    Area.objects.filter(**match_data).exclude(
-                        external_id=ext_id
-                    ).delete()
-                    match_data["external_id"] = ext_id
-                    Area.objects.update_or_create(defaults=other_data, **match_data)
+                except IntegrityError:  # There should only be one object per identifier...
+                    ext_id = other_data.pop("external_id", "")
+                    # If external id exists, we can continue deleting data.
```

Comment:
It could be useful to document why we can continue deleting data if external_id exists